generate_schedule.py

```diff
@@ -41,11 +41,30 @@ def clamp_to_gc(power):
         # helper function: make sure to stay within GC power limits
         return min(max(power, -gc.max_power), gc.max_power)
 
-    cars = {vid: [0, 0, 0] for vid in s.world_state.vehicles}
+    # collect and accumulate information about the entire fleet
+    total_vehicle_capacity = 0
+    total_desired_energy = 0
+    average_efficiency = 0
+    v2g_enabled = False
+    for v in s.world_state.vehicles.values():
+        total_vehicle_capacity += v.battery.capacity
+        total_desired_energy += v.desired_soc * v.battery.capacity
```

Comment:
Shouldn't this be "desired_soc - actual_soc"? Seems like otherwise total_desired_energy is to high

Reply:
I think its correct. I added a comment for clarification about what total_desired_energy resembles.

Reply:
Now the comment is perspicuous.